File: practice_questions/interviewbit/Tree&graphs/SumRootToLeafNumbers.py
# Definition for a  binary tree node
class TreeNode:
	def __init__(self, x):
		self.val = x
		self.left = None
		self.right = None

class Solution:
	# @param A : root node of tree
	# @return an integer
    def solve(self, root,SUM):
        if not root:
            return 0
        SUM=SUM*10+root.val
        if not root.left and not root.right:
            return SUM
        else:
            return self.solve(root.left,SUM)+self.solve(root.right,SUM)
    def sumNumbers(self, A):
        SUM=0
        return self.solve(A,SUM)%1003

File: practice_questions/interviewbit/Tree&graphs/SumRootToLeafNumbers.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {

    public int solve(TreeNode root, int sum_) {
        if (root == null) {
            return 0;
        }
        sum_ = (((sum_%1003)*10)%1003) + root.val;
        
        if (root.left ==null && root.right ==null) {
            return sum_%1003;
        }

        int l = solve(root.left, sum_)%1003;
        int r = solve(root.right, sum_)%1003;

        return l+r;
    }
    public int sumNumbers(TreeNode A) {
        int sum_ = 0;
        int ans = solve(A, sum_);
        return ans % 1003;

    }
}

